sql/insert.go

```diff
@@ -119,3 +112,20 @@ func (p *planner) processColumns(desc *structured.TableDescriptor,
 
 	return cols, nil
 }
+
+func putKeyValue(b *client.Batch, key proto.Key, val parser.Expr) error {
```

Comment:
Rather than putting the key value on the batch, I think this would be more flexible if it transformed the `val` into an `interface{}` that could be passed to any of the `Put/CPut` variations. Something like:

```
func prepareVal(val parser.Expr) interface{} {
  switch t := val.(type) {
    case parser.DBool:
      return bool(t)
    ...
  }
}
```